src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -2215,6 +2251,21 @@ def get_custom_locations_oid(cmd, cl_oid):
 
 
 def troubleshoot(cmd, client, resource_group_name, cluster_name, kube_config=None, kube_context=None, no_wait=False, tags=None):
+    # Install helm client
+    helm_client_location = install_helm_client()
+
+    release_namespace = validate_release_namespace(client, cluster_name, resource_group_name, kube_config, kube_context, helm_client_location)
+
+    # Checking the connection to kubernetes cluster.
+    # This check was added to avoid large timeouts when connecting to AAD Enabled AKS clusters
+    # if the user had not logged in.
+    check_kube_connection()
+    utils.try_list_node_fix()
+    helm_values = get_all_helm_values(release_namespace, kube_config, kube_context, helm_client_location)
+
+    if helm_values.get('global').get('isLeastPrivilegesMode') is True:
```

Comment:
least priviledge clusters could also be prone to network/dns/proxy issues. can't we just run the troubleshoot as is and we  just give a warning instead of an error that all least priviledge scenarios are not covered in troubleshoot